homeassistant/components/logi_circle/sensor.py

```diff
@@ -93,45 +77,43 @@ def extra_state_attributes(self):
         }
 
-        if self._sensor_type == "battery_level":
+        if self.entity_description.key == "battery_level":
             state[ATTR_BATTERY_CHARGING] = self._camera.charging
 
         return state
 
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        if self._sensor_type == "battery_level" and self._state is not None:
+        sensor_type = self.entity_description.key
+        if sensor_type == "battery_level" and self._attr_native_value is not None:
             return icon_for_battery_level(
-                battery_level=int(self._state), charging=False
+                battery_level=int(self._attr_native_value), charging=False
             )
-        if self._sensor_type == "recording_mode" and self._state is not None:
-            return "mdi:eye" if self._state == STATE_ON else "mdi:eye-off"
-        if self._sensor_type == "streaming_mode" and self._state is not None:
-            return "mdi:camera" if self._state == STATE_ON else "mdi:camera-off"
-        return self._icon
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the units of measurement."""
-        return SENSOR_TYPES.get(self._sensor_type)[1]
+        if sensor_type == "recording_mode" and self._attr_native_value is not None:
+            return "mdi:eye" if self._attr_native_value == STATE_ON else "mdi:eye-off"
+        if sensor_type == "streaming_mode" and self._attr_native_value is not None:
+            return (
+                "mdi:camera"
+                if self._attr_native_value == STATE_ON
+                else "mdi:camera-off"
+            )
+        return self.entity_description.icon
 
     async def async_update(self):
         """Get the latest data and updates the state."""
-        _LOGGER.debug("Pulling data from %s sensor", self._name)
+        _LOGGER.debug("Pulling data from %s sensor", self.name)
         await self._camera.update()
 
-        if self._sensor_type == "last_activity_time":
+        if self.entity_description.key == "last_activity_time":
             last_activity = await self._camera.get_last_activity(force_refresh=True)
             if last_activity is not None:
                 last_activity_time = as_local(last_activity.end_time_utc)
-                self._state = (
+                self._attr_native_value = (
                     f"{last_activity_time.hour:0>2}:{last_activity_time.minute:0>2}"
                 )
         else:
-            state = getattr(self._camera, self._sensor_type, None)
+            state = getattr(self._camera, self.entity_description.key, None)
             if isinstance(state, bool):
-                self._state = STATE_ON if state is True else STATE_OFF
-            else:
-                self._state = state
-            self._state = state
+                self._attr_native_value = STATE_ON if state is True else STATE_OFF
+            self._attr_native_value = state
```

Comment:
The last line willalways overwrite the state (even in the case of a boolean value)

Reply:
Good catch! Only from looking at the previous code, I would guess the line should be inside the else block.
```py
            if isinstance(state, bool):
                self._state = STATE_ON if state is True else STATE_OFF
             else:
                 self._state = state
             self._state = state   # This was probably a mistake
```

Reply:
Yeah this solves a bug probably 😓